File: kaola_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {

    this.findAll=function () {
        return $http.get("../brand/findAll.do")
    }

    //保存
    this.save=function (entity) {
        return $http.post("../brand/save.do",entity)
    }


    //修改
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    //查询单条
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }


    //删除
    this.dele=function (ids) {
        return $http.get("../brand/del.do?ids="+ids);
    }

    //分页查询
    this.findPage = function (page,rows) {
        return $http.get("../brand/findPage.do?page="+page+"&rows="+rows);
    }

    //查询
    this.search=function (page,rows,entity) {
        return $http.post("../brand/search.do?page="+page+"&rows="+rows,entity)
    }

    //查询品牌列表
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
})

File: kaola_sellergoods_interface/src/main/java/com/kaola/sellergoods/service/BrandService.java
package com.kaola.sellergoods.service;

import com.kaola.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<Map> selectOptionList();

    //查询所有
    public List<TbBrand> findAll();
    //分页查询
    public PageResult findPage(int page,int pageSize);
    //添加
    public void save(TbBrand tbBrand);
    //修改
    public void update(TbBrand tbBrand);
    //删除
    public void  del(Long[] id);
    //修改前的查询
    public TbBrand findOne(Long id);

    //条件搜索方法
    public PageResult search(int page,int pageSize,TbBrand brand);
}
